package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +190,73 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+M.PipePolicies = {
+	DISCARD=-1,
+	INHERIT=0,
+	CREATE=1
+}
+
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.popen3(policies, path, ...)
+	local pipes = {}
+	local intern = {}
+	local extern = {}
+
+	for fd, policy in pairs(policies) do
+		if M.PipePolicies.CREATE==policy then
```

Comment:
Please don't use Yoda conditions, they are generally confusing, and in Lua they don't even prevent mistakes, as there aren't constants.

Reply:
resolved in c1df156